compiler/plugins/target/ROCM/Dialect/ROCM/IR/ROCMAttrs.cpp

```diff
@@ -79,6 +91,362 @@ static LogicalResult handleArgmaxUkernel(
   return success();
 }
 
+static constexpr char executableObjectsAttrName[] = "hal.executable.objects";
+
+// Walks parents ops from `op` to return the nearest hal.executable.objects
```

Comment:
As discussed offline, we could move these ukernel utilities and functionality into a separate file to avoid overloading ROCMAttrs.cpp. This could be a follow-up because there is some existing argmax ukernel related logic in this file already and it would be good to move that out together.